arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,268 @@
+        let utilityTierConfiguration = domainNetwork?.tier(withName: tierName)?.traceConfiguration
+        return utilityTierConfiguration
+    }
+    
+    /// Add a default phase to the list to better showcase the sample.
+    func addDefaultPhase() {
+        guard let defaultPhase = phaseChoices.first else { return }
+        phaseSummaries.append((defaultPhase, nil))
+        tableView.insertRows(at: [IndexPath(row: phaseSummaries.endIndex - 1, section: 0)], with: .automatic)
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func resetBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        phaseSummaries.removeAll()
+        tableView.reloadData()
+    }
+    
+    @IBAction func addBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        let selectedPhases = Set(phaseSummaries.map { $0.phase })
+        let remainingPhases = phaseChoices.filter { !selectedPhases.contains($0) }
+        guard !remainingPhases.isEmpty else { return }
+        let alertController = UIAlertController(
+            title: "Add a phase to get the load report.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        remainingPhases.forEach { phase in
+            alertController.addAction(UIAlertAction(title: phase.name, style: .default) { _ in
+                self.phaseSummaries.append((phase, nil))
+                self.tableView.insertRows(at: [IndexPath(row: self.phaseSummaries.endIndex - 1, section: 0)], with: .automatic)
+            })
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    @IBAction func runBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        guard !phaseSummaries.isEmpty,
+              let expression = initialExpression,
+              let phasesNetworkAttribute = phasesNetworkAttribute,
+              let traceParameters = traceParameters else { return }
+        
+        let traceGroup = DispatchGroup()
+        SVProgressHUD.show(withStatus: "Creating load report…")
+        
+        for i in 0..<phaseSummaries.count {
+            let phase = phaseSummaries[i].phase
+            guard let phaseCode = phase.code else { continue }
+            // Create a conditional expression.
+            let phasesAttributeComparison = AGSUtilityNetworkAttributeComparison(networkAttribute: phasesNetworkAttribute, comparisonOperator: .doesNotIncludeAny, value: phaseCode)!
+            // Chain it with the base condition using an OR operator.
+            traceParameters.traceConfiguration?.traversability?.barriers = AGSUtilityTraceOrCondition(leftExpression: expression, rightExpression: phasesAttributeComparison)
+            
+            traceGroup.enter()
+            utilityNetwork.trace(with: traceParameters) { [weak self] results, _ in
```

Comment:
This is only place that `traceParameters` is actually used, right? Is there an advantage to creating it ahead of time and reusing it? Since it has to be configured each time anyway (with the trace conditions), it seems to me that it might make more sense to create the parameters here and not store them as a property. What do you think?

Reply:
Yeah the original design is pretty tangled here... So I cannot clearly tell which to hold. The design holds the trace parameters.

And it seems to me that if we don't hold the `traceParameters`, we'll instead need to hold the `traceConfiguration` and `startingLocation` to create the parameters. So I'll leave it unchanged and wait for your next review.

Reply:
To be honest, I don't know that one approach would be better than the others, so just stick with what you have.